dodo/pgp_util.py

```diff
@@ -80,9 +79,12 @@ def encrypt(msg: email.message.EmailMessage) -> email.message.EmailMessage:
             val for key, val in msg.items() if key in ['From', 'To', 'Cc']
         ])
     ]
-    recipients_keys = [k['fingerprint'] for k in Gpg.list_keys()
-                       if any(re.search(addr, u) for u in k['uids']
-                              for addr in recipients)]
+    # set strict=False for parseaddr, otherwise keys with a uid such as
+    # "mail@example.org <mail@example.org>" are skipped. These uids are considered
+    # malformed, but they occur.
+    recipients_keys = [key['fingerprint'] for key in Gpg.list_keys()
+                       if any(addr == email.utils.parseaddr(uid, strict=False)[1]
```

Comment:
```suggestion
                       if any(addr == utils.strip_email_address(uid, strict=False)
```

from here maybe?

https://github.com/akissinger/dodo/blob/194fb49523c7851bedc3ca8c11adea04830fb28d/dodo/util.py#L282-L287

Would probably have to add `strict=True` to that, so feel free to ignore if you think your approach is simpler here.

Reply:
Oh, I notice now that `strict=True` [was only added](https://docs.python.org/3/whatsnew/3.12.html#notable-changes-in-3-12-6) in Python 3.12.6, while Dodo [claims to support](https://github.com/akissinger/dodo/blob/master/setup.py#L37) older versions.

I suppose something like `kwargs = {"strict": True} if strict and sys.version_info[:3] >= (3, 12, 6) else {}` followed by `email.utils.parseaddr(e, **kwargs)` would do? Or catching `TypeError` I guess.